src/mlir/generator.h

```diff
@@ -145,126 +132,102 @@ namespace mlir::verona
+
+    /// Parses a literal.
+    llvm::Expected<ReturnValue> parseLiteral(Ast ast);
+
+    /// Parses a type.
+    Type parseType(Ast ast);
+
+    // ===================================================== MLIR Generators
     /// Generate a prototype, populating the symbol table
-    llvm::Expected<mlir::FuncOp> generateProto(
-      mlir::Location loc,
+    llvm::Expected<FuncOp> generateProto(
+      Location loc,
       llvm::StringRef name,
-      llvm::ArrayRef<mlir::Type> types,
-      llvm::ArrayRef<mlir::Type> retTy);
+      llvm::ArrayRef<Type> types,
+      llvm::ArrayRef<Type> retTy);
     /// Generates an empty function (with the first basic block)
-    llvm::Expected<mlir::FuncOp> generateEmptyFunction(
-      mlir::Location loc,
+    llvm::Expected<FuncOp> generateEmptyFunction(
+      Location loc,
       llvm::StringRef name,
       llvm::ArrayRef<llvm::StringRef> args,
-      llvm::ArrayRef<mlir::Type> types,
-      llvm::ArrayRef<mlir::Type> retTy);
-    /// Generates a conditional branch, casting to i1 if necessary
-    llvm::Error generateCondBranch(
-      mlir::Location loc,
-      mlir::Value cond,
-      mlir::Block* ifBB,
-      mlir::ValueRange ifArgs,
-      mlir::Block* elseBB,
-      mlir::ValueRange elseArgs);
-    /// Generates an unconditional loop branch (continue, creak)
-    llvm::Error
-    generateLoopBranch(mlir::Location loc, llvm::StringRef blockName);
-
-    // ======================================================= Generator Helpers
-    // These are helpers for the generators, creating simple recurrent patterns
-    // for basic constructs (casts, load/store, constants, etc).
-    // They return mlir::Values because they're not supposed to fail.
-
-    /// Generates a cast, trusting the AST did the right thing
-    mlir::Value
-    generateAutoCast(mlir::Location loc, mlir::Value value, mlir::Type type);
-    /// Generates a verona constant with opaque type
-    mlir::Value generateConstant(
-      mlir::Location loc, llvm::StringRef value, llvm::StringRef typeName);
-    /// Generates a verona alloca with specific type
-    mlir::Value generateAlloca(mlir::Location loc, mlir::Type type);
-    /// Generates a verona load (using address' type, or unkTy)
-    mlir::Value generateLoad(mlir::Location loc, mlir::Value addr);
-    /// Generates a verona store (using value's type, or unkTy)
-    mlir::Value
-    generateStore(mlir::Location loc, mlir::Value value, mlir::Value addr);
-
-    // =============================================================== Temporary
-    // These methods should disappear once the Verona dialect is more advanced.
-
-    /// Wrapper for opaque operators before we use actual Verona dialect.
-    mlir::Value genOperation(
-      mlir::Location loc,
-      llvm::StringRef name,
-      llvm::ArrayRef<mlir::Value> ops,
-      mlir::Type retTy);
+      llvm::ArrayRef<Type> types,
+      llvm::ArrayRef<Type> retTy);
+    /// Generates a call to a static/dynamic function
```

Comment:
We do, because the selector for static calls is a type, while the selector for dynamic calls is an object. If there are no selectors, then we consider the current module as the selector and just make a static call to a method on the same module.

The comment isn't describing what the function currently does, but what it's supposed to do. I'll change to reflect reality and add a FIXME.